### PR TITLE
fix(pipeline): use batched queries instead of loading full data in dbgc

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
use batched queries instead of loading full data in dbgc

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJhc3NpZ25lZSI6WyIxMDAxMjA1Il19&id=325245&iterationID=1321&pId=0&type=TASK)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |    Fix the bug that use batched queries instead of loading full data in dbgc          |
| 🇨🇳 中文    |      修复dbgc占用大量内存        |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: apistructs/pipeline.go [hunk 1]
```diff
@@ -364,9 +364,11 @@ type PipelinePageListRequest struct {
 	CountOnly bool `schema:"countOnly"` // 是否只获取 total
 
 	// internal use
-	SelectCols []string `schema:"-" ` // 需要赋值的字段列表，若不声明，则全赋值
-	AscCols    []string `schema:"ascCol"`
-	DescCols   []string `schema:"descCol"`
+	SelectCols   []string `schema:"-" ` // 需要赋值的字段列表，若不声明，则全赋值
+	AscCols      []string `schema:"ascCol"`
+	DescCols     []string `schema:"descCol"`
+	StartIDBegin uint64   `schema:"-"`
```
Comment (sfwn) [inline apistructs/pipeline.go:370]: unclear var name.
include or exclude?
refer to `gt, ge, lt, le`

File: internal/tools/pipeline/providers/dbgc/dbgc.go [hunk 2]
```diff
@@ -62,12 +62,8 @@ func (p *provider) PipelineDatabaseGC(ctx context.Context) {
 
 // doPipelineDatabaseGC query the data in the database according to req paging to perform gc
 func (p *provider) doPipelineDatabaseGC(req apistructs.PipelinePageListRequest) {
-	var pageNum = req.PageNum
-	pipelines := make([]spec.Pipeline, 0)
 	for {
```
Comment (sfwn) [inline internal/tools/pipeline/providers/dbgc/dbgc.go:65]: use `rutil.ContinueWorking`

**Review by sfwn — CHANGES_REQUESTED** [review]

File: apistructs/pipeline.go (hunk 1 above)
Comment (chengjoey) [reply apistructs/pipeline.go]: adjusted, use 'gt' 'lt'

File: internal/tools/pipeline/providers/dbgc/dbgc.go (hunk 2 above)
Comment (chengjoey) [reply internal/tools/pipeline/providers/dbgc/dbgc.go]: adjusted

File: internal/tools/pipeline/providers/dbgc/dbgc.go [hunk 3]
```diff
@@ -61,33 +61,30 @@ func (p *provider) PipelineDatabaseGC(ctx context.Context) {
...
-	for {
-		req.PageNum = pageNum
+func (p *provider) doPipelineDatabaseGC(ctx context.Context, req apistructs.PipelinePageListRequest) {
+	rutil.ContinueWorking(ctx, p.Log, func(ctx context.Context) rutil.WaitDuration {
 		result, err := p.dbClient.PageListPipelines(req)
-		pageNum += 1
 		if err != nil {
 			p.Log.Errorf("failed to compensate pipeline req: %v, err: %v", req, err)
-			continue
+			return rutil.ContinueWorkingImmediately
```
Comment (sfwn) [inline internal/tools/pipeline/providers/dbgc/dbgc.go:69]: should sleep, or you may occur error infinitely

**Review by sfwn — CHANGES_REQUESTED** [review]
pay attention to cpu load

File: internal/tools/pipeline/providers/dbgc/dbgc.go (hunk 3 above)
Comment (chengjoey) [reply internal/tools/pipeline/providers/dbgc/dbgc.go]: adjusted, sleep with default interval

**Review by sfwn — APPROVED** [review]

**Pull request merged**